inc/namespace.php

```diff
@@ -305,7 +305,12 @@ function send_trace_to_daemon( array $trace ) {
 function get_flattened_segments_from_trace( array $trace ) : array {
 	$max_size = 63 * 1024; // 63 KB, leaving room for UDP headers etc.
 	$segments = [];
-	if ( empty( $trace['subsegments'] ) || mb_strlen( json_encode( $trace ) ) < $max_size ) { // @codingStandardsIgnoreLine wp_json_encode not available.
+	$size = mb_strlen( json_encode( $trace ) ); // @codingStandardsIgnoreLine wp_json_encode not available.
+	if ( empty( $trace['subsegments'] ) || $size < $max_size ) {
+		// Truncate values until the trace is small enough to deliver.
+		if ( $size > $max_size ) {
```

Comment:
A bit confused about this conditional, will this conditional ever met? Because above, the condition is `$size < $max_size`

Reply:
It's possible for `$size > $max_size` to be true if `empty( $trace['subsegments'] )` is true, it's an `OR` operator in the outer `if` statement.